Userland/Applications/Calendar/Calendar.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Calendar.h"
+#include <AK/JsonParser.h>
+#include <LibConfig/Client.h>
+#include <LibFileSystemAccessClient/Client.h>
+
+namespace Calendar {
+
+Calendar::Calendar()
+{
+}
+
+OwnPtr<Calendar> Calendar::create()
+{
+    return adopt_own(*new Calendar());
+}
+
+ErrorOr<void> Calendar::add_event(JsonObject event)
+{
+    if (!m_events.has_value()) {
+        m_events = TRY(AK::JsonParser("[]"sv).parse()).as_array();
+    }
+
+    TRY(m_events->append(move(event)));
+
+    set_dirty(true);
+
+    on_events_change();
+
+    return {};
+}
+
+void Calendar::set_events(JsonArray events)
+{
+    m_events = events;
+    on_events_change();
+}
+
+void Calendar::save(FileSystemAccessClient::File& file)
+{
+    auto filename = file.filename().to_deprecated_string();
+    set_filename(filename);
+    set_dirty(false);
+
+    auto stream = file.release_stream();
+    if (m_events.has_value()) {
+        auto events = m_events.value();
+        (void)stream->write_some(events.to_deprecated_string().bytes());
```

Comment:
Oof. Can we not make this an `ErrorOr<void>` and not just throw away results like that?